### ISSUE TITLE
Passing initial_epoch parameter to callbacks' self.params in tf.keras.model.fit

### ISSUE THREAD

**Issue by shun-lin** [tensorflow/tensorflow#39839]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.x
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

TensorFlow Addons are developing a progress bar (TQDM Progress bar) and we receive and issue where when user specify `initialEpoch > 0` during `model.fit` and when using `tfa.callbacks.TQDMProgressBar` the progress bar will never reach 100% because in `TQDMProgressBar` code we never consider the case when user set `initialEpoch`

`
model.fit(x=X, y=y, class_weight=None, batch_size=batchSize, verbose=0, callbacks=tfa.callbacks.TQDMProgressBar(), validation_split=0.2, shuffle=True, epochs=epochCount, initial_epoch=initialEpoch)
`

https://github.com/tensorflow/addons/issues/1748 

While searching for a solution, I noticed that `initial_epoch` is never passed to Callback's `self.params` dictionary and thus making it hard to implement the feature where users set an initial epoch, thus I am asking if it would make sense for TensorFlow to pass `initial_epoch` to Callback's `self.params` for us to implement that feature. The other way around would be to ask user to specify `initial_epoch` again in the progress bar but that would not be ideal. Thank you so much for your time!

**Will this change the current api? How?**

No, but this will add another key into Callbacks' `self.params` dictionary.

**Who will benefit with this feature?**

Users of TensorFlow Addons TQDM Progress Bar and thus that may need `initial_epoch` in their custom callbacks. 

**Any Other info.**

TensorFlow Addons TQDM Progress Bar Source code: 
https://github.com/tensorflow/addons/blob/master/tensorflow_addons/callbacks/tqdm_progress_bar.py

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
